OpenFn/kit
# Finish refactoring opts

## Short Description

This PR finishes the ensureOpts refactor, removing that code entirely.

## Related issue

Fixes #115

## Implementation Details

Not all commands use the new options APIs - the simpler commands don't really need to. The point is the ensure opts stuff is gone, commands have a cleaner pattern, and we've removed some confusion. The entrypoint to the LCI is also cleaner.

There's more we can do (I'd like to remove the Opts object in favour of the individual command types), but this is good enough for now. This issue has dragged on for a very long time and fundamentally I've done  what I wanted to do.

## QA Notes

I'd love someone to do a quick run through of each command, make sure I haven't broken anything.

## Checklist before requesting a review
